plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergAlluxioCacheFileOperations.java

```diff
@@ -0,0 +1,311 @@
+import org.junit.jupiter.api.parallel.Execution;
+import org.junit.jupiter.api.parallel.ExecutionMode;
+
+import java.io.File;
+import java.nio.file.Path;
+import java.util.Optional;
+import java.util.function.Predicate;
+
+import static com.google.common.collect.ImmutableMultiset.toImmutableMultiset;
+import static com.google.inject.Scopes.SINGLETON;
+import static com.google.inject.multibindings.OptionalBinder.newOptionalBinder;
+import static io.trino.filesystem.TrackingFileSystemFactory.OperationType.INPUT_FILE_GET_LENGTH;
+import static io.trino.filesystem.TrackingFileSystemFactory.OperationType.INPUT_FILE_NEW_STREAM;
+import static io.trino.filesystem.alluxio.TestingAlluxioFileSystemCache.OperationType.CACHE_READ;
+import static io.trino.filesystem.alluxio.TestingAlluxioFileSystemCache.OperationType.EXTERNAL_READ;
+import static io.trino.plugin.base.util.Closables.closeAllSuppress;
+import static io.trino.plugin.iceberg.IcebergQueryRunner.ICEBERG_CATALOG;
+import static io.trino.plugin.iceberg.TestIcebergAlluxioCacheFileOperations.FileType.DATA;
+import static io.trino.plugin.iceberg.TestIcebergAlluxioCacheFileOperations.FileType.MANIFEST;
+import static io.trino.plugin.iceberg.TestIcebergAlluxioCacheFileOperations.FileType.METADATA_JSON;
+import static io.trino.plugin.iceberg.TestIcebergAlluxioCacheFileOperations.FileType.METASTORE;
+import static io.trino.plugin.iceberg.TestIcebergAlluxioCacheFileOperations.FileType.SNAPSHOT;
+import static io.trino.plugin.iceberg.TestIcebergAlluxioCacheFileOperations.FileType.fromFilePath;
+import static io.trino.testing.MultisetAssertions.assertMultisetsEqual;
+import static io.trino.testing.TestingSession.testSessionBuilder;
+import static java.util.Collections.nCopies;
+import static java.util.Objects.requireNonNull;
+import static java.util.stream.Collectors.toCollection;
+
+// single-threaded AccessTrackingFileSystemFactory is shared mutable state
+@Execution(ExecutionMode.SAME_THREAD)
+public class TestIcebergAlluxioCacheFileOperations
+        extends AbstractTestQueryFramework
+{
+    private TrackingFileSystemFactory trackingFileSystemFactory;
+    private TestingAlluxioFileSystemCache alluxioFileSystemCache;
+
+    @Override
+    protected DistributedQueryRunner createQueryRunner()
+            throws Exception
+    {
+        Session session = testSessionBuilder()
+                .setCatalog("iceberg")
+                .setSchema("test_schema")
+                .build();
+        DistributedQueryRunner queryRunner = DistributedQueryRunner.builder(session)
+                .build();
+        try {
+            File metastoreDirectory = queryRunner.getCoordinator().getBaseDataDir().resolve("iceberg_data").toFile().getAbsoluteFile();
+            Path dataDirectory = queryRunner.getCoordinator().getBaseDataDir().resolve("iceberg_data");
+            dataDirectory.toFile().mkdirs();
+            trackingFileSystemFactory = new TrackingFileSystemFactory(new LocalFileSystemFactory(dataDirectory));
```

Comment:
We should use OpenTelemetry tracing for this, as we plan to remove `TrackingFileSystemFactory`. It's best to construct the query runner with properties, rather than manually creating it.

Take a look at how `io.trino.plugin.hive.metastore.MetastoreInvocations` is used in tests.

Reply:
@amoghmargoor was just waiting for the PR from @wendigo to merge .. he will adjust now

Reply:
I have rebased it with @wendigo's changes